addons/mail/static/src/js/systray.js

```diff
@@ -260,15 +299,20 @@ var ActivityMenu = Widget.extend({
         } else {
             context['search_default_activities_' + data.filter] = 1;
         }
-        this.do_action({
-            type: 'ir.actions.act_window',
-            name: data.model_name,
-            res_model:  data.res_model,
-            views: [[false, 'kanban'], [false, 'form']],
-            search_view_id: [false],
-            domain: [['activity_user_id', '=', session.uid]],
-            context:context,
-        });
+        if (data.res_model === 'mail.activity') {
```

Comment:
Related to previous comment about supporting void res_model: it seems more natural to me "if data.res_model == false -> redirect to kanban view" for example.